include/layers/FCLayer.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <stdexcept>
+#include <vector>
+
+typedef double ValueType;
```

Comment:
Let's make it template, no need to make typedef

Reply:
Ok